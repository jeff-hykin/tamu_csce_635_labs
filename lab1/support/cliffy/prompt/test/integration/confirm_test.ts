import { Confirm } from "../../confirm.ts";
import { snapshotTest } from "../../../testing/snapshot.ts";

await snapshotTest({
  name: "confirm prompt",
  meta: import.meta,
  osSuffix: ["windows"],
  steps: {
    "should confirm": { stdin: ["y", "\n"] },
    "should not confirm": { stdin: ["n", "\n"] },
    "should not confirm by default": { stdin: ["", "\n"] },
  },
  async fn() {
    await Confirm.prompt({
      message: "Please confirm",
      hint: "some hint",
      default: false,
    });
  },
});
