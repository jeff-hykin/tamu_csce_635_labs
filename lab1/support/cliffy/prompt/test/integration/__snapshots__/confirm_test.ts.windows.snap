export const snapshot = {};

snapshot[`confirm prompt > should confirm 1`] = `
stdout:
"? Please confirm (y/N) » 
❯ some hint\\x1b[1A\\x1b[26G\\x1b[G\\x1b[0J
? Please confirm (y/N) » yes
❯ some hint\\x1b[1A\\x1b[27G\\x1b[G\\x1b[0J
? Please confirm (y/N) » Yes
\\x1b[?25h\\x1b[?25h"
stderr:
""
`;

snapshot[`confirm prompt > should not confirm 1`] = `
stdout:
"? Please confirm (y/N) » 
❯ some hint\\x1b[1A\\x1b[26G\\x1b[G\\x1b[0J
? Please confirm (y/N) » no
❯ some hint\\x1b[1A\\x1b[27G\\x1b[G\\x1b[0J
? Please confirm (y/N) » No
\\x1b[?25h\\x1b[?25h"
stderr:
""
`;

snapshot[`confirm prompt > should not confirm by default 1`] = `
stdout:
"? Please confirm (y/N) » 
❯ some hint\\x1b[1A\\x1b[26G\\x1b[G\\x1b[0J
? Please confirm (y/N) » No
\\x1b[?25h\\x1b[?25h"
stderr:
""
`;
